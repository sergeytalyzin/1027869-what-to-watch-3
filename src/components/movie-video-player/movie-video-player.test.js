import React from "react";
import renderer from "react-test-renderer";
import MovieVideoPlayer from "./movie-video-player.jsx";

const mock = {
  progressInPercent: 0,
  progressInSeconds: 0,
  isPlaying: false,
  title: `Some title`,
  isFullScreen: false,
};

it(`<MovieVideoPlayer /> should render trailer player`, () => {
  const tree = renderer
    .create(
        <MovieVideoPlayer
          {...mock}
          onFullScreenButtonClick={() => {}}
          onPlayButtonClick={() => {}}
          onExitFilmButtonClick={() => {}}
          type={`trailer`}
        >
          <video/>
        </MovieVideoPlayer>,
        {
          createNodeMock: () => {
            return {};
          }
        }
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`<MovieVideoPlayer /> should render full movie player`, () => {
  const tree = renderer
    .create(
        <MovieVideoPlayer
          {...mock}
          onFullScreenButtonClick={() => {}}
          onPlayButtonClick={() => {}}
          onExitFilmButtonClick={() => {}}
          type={`movie`}
        >
          <video/>
        </MovieVideoPlayer>,
        {
          createNodeMock: () => {
            return {};
          }
        }
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
