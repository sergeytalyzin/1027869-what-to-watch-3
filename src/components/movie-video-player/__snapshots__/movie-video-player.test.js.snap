// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieVideoPlayer /> should render full movie player 1`] = `
<div
  className="player"
>
  <video />
  <button
    className="player__exit"
    onClick={[Function]}
    type="button"
  >
    Exit
  </button>
  <div
    className="player__controls"
  >
    <div
      className="player__controls-row"
    >
      <div
        className="player__time"
      >
        <progress
          className="player__progress"
          max="100"
          value="0"
        />
        <div
          className="player__toggler"
          style={
            Object {
              "left": "0%",
            }
          }
        >
          Toggler
        </div>
      </div>
      <div
        className="player__time-value"
      >
        0:00:00
      </div>
    </div>
    <div
      className="player__controls-row"
    >
      <button
        className="player__play"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="19"
          viewBox="0 0 19 19"
          width="19"
        >
          <use
            xlinkHref="#play-s"
          />
        </svg>
        <span>
          Play
        </span>
      </button>
      <div
        className="player__name"
      >
        Some title
      </div>
      <button
        className="player__full-screen"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
        >
          <use
            xlinkHref="#full-screen"
          />
        </svg>
        <span>
          Full screen
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`<MovieVideoPlayer /> should render trailer player 1`] = `<video />`;
